noir-projects/aztec-nr/aztec/src/test/helpers/test_environment.nr

```diff
@@ -86,12 +89,20 @@ impl TestEnvironment {
         address
     }
 
-    fn deploy<let N: u32, let M: u32>(_self: Self, path: str<N>, name: str<M>) -> Deployer<N, M> {
-        Deployer { path, name, public_keys_hash: 0 }
+    fn deploy<let N: u32, let M: u32>(_self, path: str<N>, name: str<M>) -> Deployer<N, M> {
+        Deployer { path, name, public_keys_hash: 0, secret: 0 }
+    }
+
+    fn deploy_self<let M: u32>(_self, name: str<M>) -> Deployer<0, M> {
+        Deployer { path: "", name, public_keys_hash: 0, secret: 0 }
     }
 
-    fn deploy_self<let M: u32>(_self: Self, name: str<M>) -> Deployer<0, M> {
-        Deployer { path: "", name, public_keys_hash: 0 }
+    fn deploy_with_public_keys<let N: u32, let M: u32>(_self, path: str<N>, name: str<M>, secret: Field) -> Deployer<N, M> {
+        Deployer { path, name, public_keys_hash: 0, secret }
+    }
+
+    fn deploy_self_with_public_keys<let M: u32>(_self, name: str<M>, secret: Field) -> Deployer<0, M> {
+        Deployer { path: "", name, public_keys_hash: 0, secret }
```

Comment:
Sort of odd that these are called `with_public_keys` but they receive a secret. Maybe `with_keys` would be better?